### ISSUE TITLE
Limit temporal_extent to a single interval (in `load_collection` and `filter_temporal`)

### ISSUE THREAD

**Issue by mkadunc** [Open-EO/openeo-processes#88]
https://open-eo.github.io/openeo-api/processreference/#load_collection
https://open-eo.github.io/openeo-api/processreference/#filter_temporal

TLDR: The usage with two `time` values (without date) is complicated to understand and implement, poorly defined in the text, and does not have a clear use case.

Longer version: There are two possible ways to understand the time-only case (we most likely had the first in mind when writing the process description):

1. the extent is unlimited and filtering happens only on the time-of-day part of the collection's time dimension. This is tricky to implement efficiently, and might cause huge collections to be returned.

2. the time-only specification is only valid on collections whose temporal dimension encodes just time-of-day, not date (we don't have any examples, but they could exist). This makes sense, but raises the question of what we allow as `temporal` dimension.

The problems with the current proposal:

- spatial extent limits loading to a single connected sub-cube of the collection; complicating the temporal extent parameter to allow fragmented disconnected subcubes feels unnecessary

- It is not clear which part of RFC 3339 the time-only part refers to (example implies `full-time`, but it's not clear whether the offset part is mandatory or not)

- the description mentions *the* interval or *an* interval, but the time-only option specifies an infinite set of temporal intervals

-  even if selecting by time-of-day is found to be useful, one cannot use `load_collection` to both limit the temporal extent to e.g. a single year, and select only those scenes that were taken at a specific time-of-day - in practice, users will most likely filter by date first (load collection for 2018) and by time-of-day afterwards (select only scenes which were taken in the morning)

- with cyclical nature of time-of-day, it's reasonable to expect the right point of the interval to be numerically smaller than the left point (e.g. `20:00:00+12:00` .. `08:00:00+12:00`), descriptions should be consistent with this

**Comment by jdries** [comment]
There's also this issue: https://github.com/Open-EO/openeo-processes/issues/39
Which we could solve by allowing multiple intervals in filter_temporal.

**Comment by m-mohr** [comment]
Telco: I'll discuss with the meteorologists whether that's useful for them. If not, we'll remove the time-only filter from load_collection and leave it for filter_temporal. A two step approach while loading the data makes sense anyway as usually you'd load based on the date range and then filter based on the time.

**Comment by m-mohr** [comment]
I'm quoting Julia Wagemann here:
> In der Meteorologie sind diese Anfragen üblich für Vorhersagedaten. Für Klimadaten (z.B. Reanalysedaten) ist es weniger üblich, weil man eher an durchschnittlichen Informationen über längere Zeiträume interessiert ist. Da Vorhersagedaten bisher noch nicht „Open Data“, kann diese Funktionalität wahrscheinlich erstmal nach hinten verschoben werden.

It basically says that there are some use cases for forecast data, but due to the lack of open data in that fields it's probably not very important yet. So to find a compromise here, I'd suggest to remove it from load_collection to make this process easier to implement and leave it in the more dedicated process filter_temporal and clarify the documentation there.

**Comment by m-mohr** [comment]
I thought a bit more about it and think the "filter by time of day" use case would deserve its own process as it's different enough to warrant it. It probably makes implementation and documentation easier, too.

**Comment by m-mohr** [comment]
Merged.
